File: library/src/main/java/com/pengrad/telegrambot/model/request/InlineQueryResultCachedPhoto.java
package com.pengrad.telegrambot.model.request;

import java.io.Serializable;

/**
 * Stas Parshin
 * 06 May 2016
 */
public class InlineQueryResultCachedPhoto extends InlineQueryResult<InlineQueryResultCachedPhoto> implements Serializable {
    private final static long serialVersionUID = 0L;

    private String photo_file_id;

    private String title;
    private String description;
    private String caption;
    private String parse_mode;

    public InlineQueryResultCachedPhoto(String id, String fileId) {
        super("photo", id);
        this.photo_file_id = fileId;
    }

    public InlineQueryResultCachedPhoto title(String title) {
        this.title = title;
        return this;
    }

    public InlineQueryResultCachedPhoto description(String description) {
        this.description = description;
        return this;
    }

    public InlineQueryResultCachedPhoto caption(String caption) {
        this.caption = caption;
        return this;
    }

    public InlineQueryResultCachedPhoto parseMode(ParseMode parseMode) {
        this.parse_mode = parseMode.name();
        return this;
    }
}

File: core/src/commonMain/kotlin/com/pengrad/telegrambot/model/request/InlineQueryResultCachedPhoto.kt
package com.pengrad.telegrambot.model.request

import java.io.Serializable

/**
 * Stas Parshin
 * 06 May 2016
 */
class InlineQueryResultCachedPhoto(id: String, private val photo_file_id: String) : InlineQueryResult<InlineQueryResultCachedPhoto>("photo", id), Serializable {

    private var title: String? = null
    private var description: String? = null
    private var caption: String? = null
    private var parse_mode: String? = null

    fun title(title: String): InlineQueryResultCachedPhoto {
        this.title = title
        return this
    }

    fun description(description: String): InlineQueryResultCachedPhoto {
        this.description = description
        return this
    }

    fun caption(caption: String): InlineQueryResultCachedPhoto {
        this.caption = caption
        return this
    }

    fun parseMode(parseMode: ParseMode): InlineQueryResultCachedPhoto {
        this.parse_mode = parseMode.name
        return this
    }

    companion object {
        private const val serialVersionUID = 0L
    }
}
